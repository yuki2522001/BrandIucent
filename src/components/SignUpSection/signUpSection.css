.sectionSignUp {
  display: flex;
  background: var(--bg-light);
}

.sectionSignUp__content {
  margin-top: 167px;
}

.sectionSignUp__info {
  margin: 89px 0 29px;
}

.sectionSignUp__address > button {
  margin-left: 10px;
}