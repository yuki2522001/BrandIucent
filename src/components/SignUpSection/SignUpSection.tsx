import React, { memo } from "react";
import url from "@assets/images/backgrounds/vegetables-fruits.png";
import Text from "@components/common/Text/Text";
import Input from "@components/common/Input/Input";
import Button from "@components/common/Button/Button";
import "./signUpSection.css";

const SignUpSection: React.FC = () => {
  return (
    <section data-testid="signUp-section" className="sectionSignUp">
      <figure>
        <img src={url} alt="This is image vegetables-fruits" />
      </figure>
      <div className="sectionSignUp__content">
        <Text
          text="Lorem ipsum dolor sit amet, consectetur adipiscing elit."
          size="medium"
        />
        <div className="sectionSignUp__info">
          <Text
            text="Subscribe to get 10% Discount and promotion"
            size="normal"
          />
        </div>
        <div className="sectionSignUp__address">
          <Input type="text" placeholder="Email address" />
          <Button text="Sign Up" type="large" />
        </div>
      </div>
    </section>
  );
};

export default SignUpSection;
