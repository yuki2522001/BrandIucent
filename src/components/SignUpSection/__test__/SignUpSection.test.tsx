import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import SignUpSection from "../SignUpSection";

describe("SignUpSection component", () => {
  test("should render component SignUpSection", () => {
    const { getByTestId } = render(<SignUpSection />);
    expect(getByTestId("signUp-section")).toBeInTheDocument();
  });

  test("matches snapshot", () => {
    const { asFragment } = render(<SignUpSection />);
    expect(asFragment()).toMatchSnapshot();
  });
});