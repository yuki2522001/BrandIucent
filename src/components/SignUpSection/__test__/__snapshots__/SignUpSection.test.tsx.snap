// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUpSection component matches snapshot 1`] = `
<DocumentFragment>
  <section
    class="sectionSignUp"
    data-testid="signUp-section"
  >
    <figure>
      <img
        alt="This is image vegetables-fruits"
      />
    </figure>
    <div
      class="sectionSignUp__content"
    >
      <p
        class="text--medium"
        data-testid="text"
      >
        Lorem ipsum dolor sit amet, consectetur adipiscing elit.
      </p>
      <div
        class="sectionSignUp__info"
      >
        <p
          class="text--normal"
          data-testid="text"
        >
          Subscribe to get 10% Discount and promotion
        </p>
      </div>
      <div
        class="sectionSignUp__address"
      >
        <input
          class="input__value"
          data-testid="input-value"
          placeholder="Email address"
          type="text"
        />
        <button
          class="btn btn--large"
        >
          Sign Up
        </button>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
